### PR TITLE
project initialize command

### PR DESCRIPTION
**Pull request by abdalasif**
@pthreat

### CODE REVIEW

File: src/LDL/DevTools/Project.php [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    {
+        $this->__initialize();
+    }
+    
+    /**
+     * install project
+     *
+     * @return void
+     */
+    private function __initialize(): void
```
Comment (pthreat) [inline src/LDL/DevTools/Project.php:87]: Code inspections will bitch about "Only magic methods are supposed to have double lower dashes"

Comment (abdalasif) [reply src/LDL/DevTools/Project.php:87]: 😆 it supposed to be single.

File: src/LDL/DevTools/Project.php [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    }
+    
+    /**
+     * check or install composer
+     *
+     * @return void
+     */
+    private function installComposer(): void
+    {
+        $command = Command::run("composer");
```
Comment (pthreat) [inline src/LDL/DevTools/Project.php:159]: Single quotes instead of double quotes :)

File: src/LDL/DevTools/Project.php [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        if ($command->failed) {
+            $this->composerPath = __DIR__ . "/../../../bin/composer";
+            
+            if(is_writable($this->composerPath)) {
+                throw new Exception("ldl-dev-tools/bin directory is not writable");
+            }
+    
+            $command = Command::run("
+                php -r \"copy('https://getcomposer.org/installer', '{$this->composerPath}-setup.php');\" && 
+                php -r \"if (hash_file('sha384', '{$this->composerPath}-setup.php') !== '906a84df04cea2aa72f40b5f787e49f22d4c2f19492ac310e8cba5b96ac8b64115ac402c8cd292b8a03482574915d1a8') { echo 'Installer corrupt'; unlink('{$this->composerPath}-setup.php'); }\"
```
Comment (pthreat) [inline src/LDL/DevTools/Project.php:170]: Hash *may* change when the switch to a diff version

File: src/LDL/DevTools/Project.php [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+        if ($command->failed) {
+            $this->composerPath = __DIR__ . "/../../../bin/composer";
+            
+            if(is_writable($this->composerPath)) {
+                throw new Exception("ldl-dev-tools/bin directory is not writable");
+            }
+    
+            $command = Command::run("
+                php -r \"copy('https://getcomposer.org/installer', '{$this->composerPath}-setup.php');\" &&
```
Comment (pthreat) [inline src/LDL/DevTools/Project.php:169]: This should be Command::run('php') right ?

File: src/LDL/DevTools/Project.php [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    /**
+     * check or install composer
+     *
+     * @return void
+     */
+    private function installComposer(): void
+    {
+        $command = Command::run("composer");
+
+        if ($command->failed) {
```
Comment (pthreat) [inline src/LDL/DevTools/Project.php:161]: It's shorter if you say: 

```
if(!$command->failed) {
   return;
}
```

File: src/LDL/DevTools/Project.php [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+    }
+    
+    /**
+     * install cs-fixer
+     *
+     * @return void
+     */
+    private function installCsFixer(): void
+    {
+        $gitInit = $this->newProject ? "&& git init" : "";
```
Comment (pthreat) [inline src/LDL/DevTools/Project.php:130]: Also should be Command::run

**Comment by abdalasif** [comment]
@pthreat  updated PR for exception and detailed error messages.

**Pull request closed without merging**